### PR TITLE
Add config class for data sets

### PR DESCRIPTION
**Pull request by einarmo**
In order to align with python utils, it's nice to have a config class with logic for fetching data set ID.

### CODE REVIEW

**Review by ozangoktan — COMMENTED** [review]
Generally looks good, but I am thinking whether client-aware logic belongs under configuration section. 

Would this functionality fit better in a place like Cognite.Extensions, as an extension for DataSetsResource?

File: ExtractorUtils/Configuration/BaseConfig.cs [hunk 1]
```diff
@@ -364,5 +369,66 @@ public class RedisConfig
...
+        /// </summary>
+        /// <param name="client">Client to use for retrieval.</param>
+        /// <param name="token">Cancellation token</param>
+        /// <returns>Dataset ID if configured</returns>
+        public async Task<long?> GetDataSetId(Client client, CancellationToken token)
+        {
+            if (Id != null) return Id.Value;
+            if (ExternalId == null) return null;
+            if (client == null) throw new ArgumentNullException(nameof(client));
+
```
[the remark this answers is not in the document]
Comment (ozangoktan) [reply ExtractorUtils/Configuration/BaseConfig.cs:427]: For clarity, the same behaviour/checks seems to exist on the following method call.

**Comment by einarmo** [comment]
> Generally looks good, but I am thinking whether client-aware logic belongs under configuration section.
> 
> Would this functionality fit better in a place like Cognite.Extensions, as an extension for DataSetsResource?

Maybe? I did it this way because that's how the python utils does it... I'll give it some thought.

**Review by ozangoktan — APPROVED** [review]
LGTM! 🚀

**Pull request merged**